desktop/libs/notebook/src/notebook/connectors/spark_shell.py

```diff
@@ -151,21 +151,32 @@ class SparkApi(Api):
   STANDALONE_JOB_RE = re.compile("Got job (\d+)")
 
   @staticmethod
-  def get_properties():
-    return SparkConfiguration.PROPERTIES
+  def get_props(properties=None):
+    props = dict([(p['name'], p['value']) for p in SparkConfiguration.PROPERTIES])
+    if properties is not None:
+      props.update(dict([(p['name'], p['value']) for p in properties]))
 
-  def create_session(self, lang='scala', properties=None):
-    if not properties:
-      config = None
-      if USE_DEFAULT_CONFIGURATION.get():
-        config = DefaultConfiguration.objects.get_configuration_for_user(app='spark', user=self.user)
+    return props
 
-      if config is not None:
-        properties = config.properties_list
-      else:
-        properties = self.get_properties()
+  @staticmethod
+  def get_properties(props=None):
+    properties = list()
+    for p in SparkConfiguration.PROPERTIES:
+      properties.append(p.copy())
+
+    if props is not None:
+      for p in properties:
+        if p['name'] in props:
+          p['value'] = props[p['name']]
```

Comment:
Is there a way to have only one get_properties() method?

Reply:
There is double transformation in function `create_session`:
- convert `properties`(type list) to props(type dict). Dict is required for function `create_session`(livy) (where `props` will be serialized to json and send to livy).
- convert `props`(type dict) back to `properties`(type list). Required in function returning.

I didn't investigate how returning object `properties` are used later and is it possible to return `props` and don't make back transformation.

Reply:
I will rename method `get_properties()` to `to_properties()`